scripts/miniapps/SNAP.md

```diff
@@ -0,0 +1,136 @@
+# Instruction for building and running SNAP
+
+Implementation: Fortran, MPI+OpenMPI, uses MPI_THREAD_SERIALIZE
+
+Useful links:
+- GitHub repository: https://github.com/losalamos/SNAP
+- Useful build and run info: http://www.nersc.gov/users/computational-systems/cori/nersc-8-procurement/trinity-nersc-8-rfp/nersc-8-trinity-benchmarks/snap
+
+1. Download from the GitHub repository
+   ```
+% git clone https://github.com/losalamos/SNAP.git
+   ```
+
+   README.md in the top level directory contains some useful information
+
+2.  Load necessary PrgEnv module, set PATH and LD_LIBRARY_PATH and set
+recommended environment variables, e.g. for MPICH, etc.
+
+3. Modify the Makefile in the `src` directory as necessary.  At most,
+three changes may be needed.
+   ```
+   #FORTRAN = mpif90
+   #For Gnu and Intel compilers, mpifort works
+   FORTRAN = mpifort
+   ```
+
+   Select one target. When using GNU, choose `gsnap`; when using Intel
+   for KNL, choose `ksnap`; when using Intel for non-KNL choose `isnap`.
+   ```
+   TARGET = gsnap
+   #TARGET = isnap
+   #TARGET = ksnap
+   ```
+
+   If KNL is targeted, specify automatic cpu dispatch so that
+   executable can run on KNL and non-KNL.
+   ```
+   #FFLAGS = -O3 $(OMPFLAG) -xmic-avx512 -ip -align array64byte -qopt-streaming-cache-evict=0 -qno-opt-dynamic-align -fp-model fast -fp-speculation fast -fno-alias -fno-fnalias
+   FFLAGS = -O3 $(OMPFLAG) -axMIC-AVX512 -ip -align array64byte -qopt-streaming-cache-evict=0 -qno-opt-dynamic-align -fp-model fast -fp-speculation fast -fno-alias -fno-fnalias
+   ```
```

Comment:
The important bit here is the `-ax` which enables the resultant binary to be run on both MIC and Xeon (as stated above).